### ISSUE TITLE
Generate type not offered for complex pattern type.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#9089]
```c#
            if (x is Blah { X is int i })
            {
            }
```

No code action offered to generate a type for 'Blah'.

**Comment by gafter** [comment]
@cyrus It appears many code actions are broken in future. When I try (control-.) to get code actions for

```cs
        Blah foo = null;
```

I get an exception

![image](https://cloud.githubusercontent.com/assets/3804346/13411219/d70b7fd2-def0-11e5-80aa-a18d07732a39.png)

resulting in an error in VS

![image](https://cloud.githubusercontent.com/assets/3804346/13411234/e65fbc32-def0-11e5-8e45-da7f7fa1fa56.png)

The same thing happens for undefined types in pattern-matching, so I can't see any pattern-matching-specific issue here.

**Comment by gafter** [comment]
OK, I rebuilt and restarted the VS from the `future` branch, and it now works, offering to generate the type.

**Comment by gafter** [comment]
No, it does not work when the patterns feature is enabled.

**Comment by CyrusNajmabadi** [comment]
Fixed with 38f8b8fad630194608d75dc6aa240e29c2311f83
